src/QsCompiler/DataStructures/SyntaxExtensions.fs

```diff
@@ -239,14 +224,23 @@ type TypedExpression with
         | NEQ (lhs, rhs)
         | RangeLiteral (lhs, rhs)
         | ArrayItem (lhs, rhs)
-        | CallLikeExpression (lhs, rhs) -> upcast [ lhs; rhs ]
+        | CallLikeExpression (lhs, rhs) -> seq [ lhs; rhs ]
         | CopyAndUpdate (ex1, ex2, ex3)
-        | CONDITIONAL (ex1, ex2, ex3) -> upcast [ ex1; ex2; ex3 ]
+        | CONDITIONAL (ex1, ex2, ex3) -> seq [ ex1; ex2; ex3 ]
         | StringLiteral (_, items)
         | ValueTuple items
-        | ValueArray items -> upcast items
-        | kind when TypedExpression.IsAtomic kind -> Seq.empty
-        | _ -> NotImplementedException "missing implementation for the given expression kind" |> raise
+        | ValueArray items -> seq items
+        | SizedArray (value, size) -> seq [ value; size ]
+        | UnitValue
+        | Identifier _
+        | IntLiteral _
+        | BigIntLiteral _
+        | DoubleLiteral _
+        | BoolLiteral _
+        | ResultLiteral _
+        | PauliLiteral _
+        | MissingExpr
+        | InvalidExpr -> Seq.empty
```

Comment:
Was it intentional to remove the `NotImplementedException "missing implementation for the given expression kind"` when no expression kind is matched?

Reply:
Yes, the match is exhaustive now. There are no unhandled expression kinds.